core.go

```diff
@@ -167,3 +167,21 @@ func (gateway *CoreGateway) DirectRefund(orderID string, req *RefundReq) (Respon
 
 	return resp, nil
 }
+
+// Subscribe : Perform transaction using subscriptions
+func (gateway *CoreGateway) Subscribe(req *Subscribe) (Subscribe, error) {
```

Comment:
Is this func return `Subscribe` struct, same struct as the request itself?
For now, all the other func are returning `Response` struct, (which might not be ideal design) but for now consistency is more important to prevent introducing inconsistent new function. Which can mix things up and may hurt long term maintenance.

Reply:
also aware of this, should I make it on another file instead like `iris.go`? @rizdaprasetya

Reply:
@ardityawahyu
Let's follow the other response, response is being defined on:
https://github.com/veritrans/go-midtrans/blob/master/response.go
So the response struct is `SubscriptionResponse`.

The request struct is `SubscriptionRequest` on:
https://github.com/veritrans/go-midtrans/blob/master/request.go